frontend/src/utilities/addTypesToK8sListedResources.ts

```diff
@@ -0,0 +1,14 @@
+import { K8sResourceCommon } from '@openshift/dynamic-plugin-sdk-utils';
+import { K8sResourceListResult } from '~/k8sTypes';
+
+export const addTypesToK8sListedResources = <TResource extends Partial<K8sResourceCommon>>(
+  response: K8sResourceListResult<TResource>,
+  kind: string,
+): K8sResourceListResult<TResource> => ({
+  ...response,
+  items: response.items.map((i) => ({
+    ...i,
+    apiVersion: response.apiVersion,
+    kind: kind,
```

Comment:
Nit
```suggestion
    kind,
```

Reply:
@DaoDaoNoCode log a tech debt please... Nits like this should be converted into Tech Debt eslinter issues.

Reply:
Didn't know linter can even do this... Will log an issue tomorrow.

Reply:
Tech-debt created: https://github.com/opendatahub-io/odh-dashboard/issues/1883